src/plugins/integrations/AWSIntegration.ts

```diff
@@ -19,349 +19,551 @@ import InvocationSupport from '../support/InvocationSupport';
-              }
+    static injectTraceLink(span: ThundraSpan, req: any, config: any): void {
+        if (span.getTag(SpanTags.TRACE_LINKS) || !req) {
+            return;
+        }
+        const region = _.get(req, 'service.config.region', '');
+        const serviceEndpoint = _.get(req, 'service.config.endpoint', '');
+        const serviceName = Utils.getServiceName(serviceEndpoint as string);
+        const operationName = req.operation;
+        const response = req.response;
+        const params = Object.assign({}, req.params);
+        let traceLinks: any[] = [];
+
+        if (serviceName === 'dynamodb') {
+            const tableName = Utils.getDynamoDBTableName(req);
+            let timestamp: number;
+            if (_.has(response, 'httpResponse.headers.date')) {
+                timestamp = Date.parse(response.httpResponse.headers.date) / 1000;
+            } else {
+                timestamp = Math.floor(Date.now() / 1000) - 1;
             }
-          } else if (serviceName === 'dynamodb') {
-            const statementType = DynamoDBRequestTypes[operationName];
-            const tableName = Utils.getDynamoDBTableName(request);
-
-            const spanName = koalas(tableName, AWS_SERVICE_REQUEST);
-
-            activeSpan = tracer._startSpan(spanName, {
-              childOf: parentSpan,
-              domainName: DomainNames.DB,
-              className: ClassNames.DYNAMODB,
-              disableActiveStart: true,
-              tags: {
-                [DB_TYPE]: DBTypes.DYNAMODB,
-                [DB_INSTANCE]: serviceEndpoint,
-                [DBTags.DB_STATEMENT_TYPE]: statementType ? statementType : '',
-                [SpanTags.OPERATION_TYPE]: statementType ? statementType : '',
-                [SpanTags.SPAN_TYPE]: SpanTypes.AWS_DYNAMO,
-                [AwsSDKTags.REQUEST_NAME]: operationName,
-                [DBTags.DB_STATEMENT]: config.maskDynamoDBStatement ? undefined : request.params,
-              },
-            });
-
-            if (statementType) {
-              activeSpan.setTag(SpanTags.TRIGGER_OPERATION_NAMES, [functionName]);
-              activeSpan.setTag(SpanTags.TOPOLOGY_VERTEX, true);
-              activeSpan.setTag(SpanTags.TRIGGER_DOMAIN_NAME, LAMBDA_APPLICATION_DOMAIN_NAME);
-              activeSpan.setTag(SpanTags.TRIGGER_CLASS_NAME, LAMBDA_APPLICATION_CLASS_NAME);
-              activeSpan.setTag(AwsDynamoTags.TABLE_NAME, tableName);
+
+            if (operationName === 'putItem') {
+                traceLinks = AWSIntegration.generateDynamoTraceLinks(params.Item, 'PUT', tableName, region, timestamp);
+            } else if (operationName === 'updateItem') {
+                traceLinks = AWSIntegration.generateDynamoTraceLinks(params.Key, 'UPDATE', tableName, region, timestamp);
+            } else if (operationName === 'deleteItem') {
+                if (config.dynamoDBTraceInjectionEnabled && _.has(response, 'data.Attributes.x-thundra-span-id.S')) {
```

Comment:
Maybe this should check  (config.dynamoDBTraceInjectionEnabled && _.has(response, 'data.Attributes.x-thundra-span-id')) and set the same as well?